crates/bevy_ecs/src/system/function_system.rs

```diff
@@ -196,6 +196,8 @@ impl<Param: SystemParam> SystemState<Param> {
         self.world_id == world.id()
     }
 
+    /// For all current [`Archetype`]s present in the [`World`],
```

Comment:
```suggestion
    /// For any newly-added [`Archetype`]s in the [`World`],
```

Reply:
It seems to be that it's actually for all current valid archetypes?
(Call to `world.archetypes()`)

Reply:
It only calls `new_archetype` for the archetypes that have been added since `self.archetype_generation`.